### ISSUE TITLE
Cannot connect to MATLAB

### ISSUE THREAD

**Issue by shan-donahue** [OpenNFT/OpenNFT#81]
hello,
I had a previous issue which was solved and I can now run OpenNFT I am now having problems when initializing the application I for some reason cannot connect to MATLAB. I am still running on a MAC OS with python 3.6 and MATLAB R2020a. I will copy and paste the error message below when initializing: 

matlab.engine.EngineError: Unable to connect to MATLAB session 'MATLAB_NFB_MAIN_15576'.
2020-07-10 08:26:56 | ERROR     | runmatlab:connect_to_matlab - Cannot connect to Matlab "MATLAB_NFB_MAIN"

I hope to hear back thank you in advance.

**Comment by espdev** [comment]
@shan-donahue could you try to perform the following instructions?

1. Open "Python Console" in PyCharm with OpenNFT project (or just run python from created "opennft" virtualenv)
2.. Run the code in the console:
```
import matlab.engine as me
e = me.connect_matlab()
e.version()
e.matlabroot()
```

Do these commands work? Do you see the version and path or get `matlab.engine.EngineError`?

**Comment by shan-donahue** [comment]
Hello I just ran those and copy and pasted from the console: 
>>> import matlab.engine as me
>>> e = me.connect_matlab()
>>> e.version()
'9.8.0.1396136 (R2020a) Update 3'
>>> e.matlabroot()
'/Applications/MATLAB_R2020a.app'
>>>

**Comment by espdev** [comment]
Thanks. At least Matlab Engine for Python is working. That is interesting why the almost same code is not working inside OpenNFT in your case.

Could you try to run Matlab sessions via `runmatlab`?
```
python -m opennft.runmatlab
```

**Comment by ykoush** [comment]
As far as I understand this is an old issue for MacOS, and seems nothing has been changed between Matlab and Python since 2016...
Please see Manual on this issue: https://github.com/OpenNFT/opennft.github.io/blob/master/OpenNFT_Manual_v1.0.pdf
And, #37 E.g., we have macos_startmatlab.sh and/or manual launch without .sh.

**Comment by espdev** [comment]
@ykoush 
It seems to me now we can use run_matlab (see my previous message) instead of macos_startmatlab.sh

**Comment by ykoush** [comment]
Not sure if run_matlab is incorporating MacOS features mentioned in the manual and #37 on starting via Matlab engine. I'd try both ways, or actually in three possible ways.

**Comment by shan-donahue** [comment]
Hello, Thank you for the suggestions I am trying each of them for : 
python -m opennft.runmatlab command being ran I still get the same connection error with MATLAB still opening 
I also went in and changed the macos_startmatlab.sh file as according to the #37 issue to be now MATLABR2020a rather than the R2016 to try and fix that issue an I still have the connection error and it will now give me: 
2020-07-13 08:35:59 | ERROR     | runmatlab:connect_to_matlab - Cannot connect to Matlab "MATLAB_NFB_MAIN"
2020-07-13 08:36:59 | ERROR     | mlproc:_start_matlab - Cannot terminate Matlab process correctly
with a SystemError: MATLAB process cannot be terminated.

**Comment by ykoush** [comment]
Did you try to run matlab processes manually, line by line in the terminal? May be @tiborauer or @sweetneuron could help you with most recent macos set.

**Comment by eniluap** [comment]
Hello, I am having the same issue with Matlab 2018a (running under Ubuntu 16). Did you solve this? Thank you

**Comment by ykoush** [comment]
Sorry, still work in progress by @da0c, @espdev  and @TheAmid now. Not sure, but may be @sweetneuron  or @tiborauer also had a chance to work on it.
But I guess you tried all possible solutions noted so far? I mean, as noted above https://github.com/OpenNFT/OpenNFT/issues/81#issuecomment-656799280
Initially, we were working on Matlabs 2016-2017 (and made our solutions for MacOS), but since many users are using MR Scanner consoles operated on Win, we didn't have a possibility to track on evolved version conflicts between Python and Matlab on MacOS/Ubuntu, etc
This issue doesn't have anything to do with OpenNFT itself, but with Matlab parallel processes working under Python, which could be reproduced/solved without OpenNFT first. I hope we can resolve it soon, or any other users could help.

**Comment by tiborauer** [comment]
I have read in the first comment that @shan-donahue uses python 3.6. I use 3.5 (on Win) and I do not have such an issue.

**Comment by eniluap** [comment]
Hello, it finally worked on linux (and Matlab 2018b; Python 3.6) with #37 solution. Thanks!

**Comment by TheAmid** [comment]
Hello.

Solution from #37 works and the script should be run before the project with -desktop parameter. Here is the example:

/Applications/Polyspace/R2020a/bin/matlab -desktop -r "matlab.engine.shareEngine('MATLAB_NFB_MAIN_00001')" > /dev/null 2>&1 &
/Applications/Polyspace/R2020a/bin/matlab -desktop -r "matlab.engine.shareEngine('MATLAB_NFB_PTB_00001')" > /dev/null 2>&1 &
/Applications/Polyspace/R2020a/bin/matlab -desktop -r "matlab.engine.shareEngine('MATLAB_NFB_SPM_00001')" > /dev/null 2>&1 &

But should specify the MATLAB directory according the version you use. "Initialize" button will open new session of MATLAB, but 
connect to the ones from the script.

**Comment by espdev** [comment]
Finally fixed in #90
